src/main/java/baritone/process/CraftingProcess.java

```diff
@@ -0,0 +1,430 @@
+            Goal[] goals = ((GoalComposite) goal).goals();
+            for (Goal goal : goals) {
+                if (goal.isInGoal(ctx.playerFeet()) && goal instanceof GoalGetToBlock) {
+                    bp = ((GoalGetToBlock) goal).getGoalPos();
+                    break;
+                }
+            }
+        } else if (goal instanceof GoalGetToBlock) {
+            bp = ((GoalGetToBlock) goal).getGoalPos();
+        }
+
+        if (bp != null) {
+            Optional<Rotation> reachable = RotationUtils.reachable(ctx, bp, ctx.playerController().getBlockReachDistance());
+            if (reachable.isPresent()) {
+                baritone.getLookBehavior().updateTarget(reachable.get(), true);
+                if (bp.equals(ctx.getSelectedBlock().orElse(null))) {
+                    baritone.getInputOverrideHandler().setInputForceState(Input.CLICK_RIGHT, true);
+                    if (!(ctx.player().openContainer instanceof ContainerPlayer)) {
+                        baritone.getInputOverrideHandler().clearAllKeys();
+                    }
+                }
+            }
+        }
+    }
+
+    private void placeCraftingTable() {
+        Optional<Placement> toPlace = searchPlacement();
+
+        if (toPlace.isPresent()) {
+            Rotation rot = toPlace.get().rot;
+            baritone.getLookBehavior().updateTarget(rot, true);
+            baritone.getInventoryBehavior().throwaway(true, this::isCraftingTable);
+            baritone.getInputOverrideHandler().setInputForceState(Input.SNEAK, true);
+            if ((ctx.isLookingAt(toPlace.get().placeAgainst) && ctx.objectMouseOver().sideHit.equals(toPlace.get().side)) || ctx.playerRotations().isReallyCloseTo(rot)) {
+                baritone.getInputOverrideHandler().setInputForceState(Input.CLICK_RIGHT, true);
+                //now that the table is placed path to it
+                placeCraftingTable = false;
+                goal = new GoalGetToBlock(toPlace.get().placeAgainst.offset(toPlace.get().side));
+            }
+        } else {
+            goal = new GoalRunAway(5, ctx.playerFeet());
+        }
+    }
+
+    private Optional<Placement> searchPlacement() {
+        for (BetterBlockPos bbp : blockPosListSortedByDistanceFromCenter()) {
+            if (MovementHelper.isReplaceable(bbp.x, bbp.y, bbp.z, baritone.bsi.get0(bbp.x, bbp.y, bbp.z), baritone.bsi)) {
+                if (bbp.y == ctx.playerHead().y && baritone.bsi.get0(bbp.x, bbp.y + 1, bbp.z).getBlock() == Blocks.AIR) {
+                    continue;
+                }
+                Optional<Placement> opt = possibleToPlace(Blocks.CRAFTING_TABLE.getDefaultState(), bbp.x, bbp.y, bbp.z, baritone.bsi);
+                if (opt.isPresent()) {
+                    return opt;
+                }
+            }
+        }
+        return Optional.empty();
+    }
+
+    private List<BetterBlockPos> blockPosListSortedByDistanceFromCenter() {
+        //this is so stupid idk why im doing this
```

Comment:
Did you try just creating the full list and then sorting it? If that's too slow (doubt it) there's ways to generate the positions in order.